File: greedy/number_of_laser_beams_in_a_bank.py
from typing import List


class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        ans = 0
        prev_devices = bank[0].count("1")
        for i in range(1, len(bank)):
            cur_devices = bank[i].count("1")
            if cur_devices == 0:
                continue
            ans += prev_devices * cur_devices
            prev_devices = cur_devices

        return ans

File: python/greedy/number_of_laser_beams_in_a_bank.go
package main

import (
	"strings"
)

func numberOfBeams(bank []string) int {
	ans := 0
	prevDevices := strings.Count(bank[0], "1")

	for i := 1; i < len(bank); i++ {
		curDevices := strings.Count(bank[i], "1")
		if curDevices == 0 {
			continue
		}
		ans += prevDevices * curDevices
		prevDevices = curDevices
	}

	return ans
}
